### PR TITLE
make vapor take collection as input

### PR DESCRIPTION
**Pull request by kciy**
Vapor should take pairs and collections of fastq files

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR updates an existing tool or tool collection

### CODE REVIEW

File: tools/vapor/vapor.xml [hunk 1]
```diff
@@ -70,29 +93,45 @@
...
             <param name="fasta_file" value="HA_sample.fa" />
-            <param name="fastq_file" ftype="fastqsanger.gz" value="test_reads.fastqsanger.gz" />
             <output name="output_scores" file="output1.tab" />
         </test>
-        <test expect_num_outputs="1"><!-- Test 4: opt -->
+        <test expect_num_outputs="1">
+            <conditional name="fastq_input">
+                <param name="fastq_input_selector" value="single" />
+                <param name="fastq_single" ftype="fastqsanger.gz" value="test_reads.fastqsanger.gz" />
+            </conditional>
```
Comment (wm75) [inline tools/vapor/vapor.xml:125]: This one seems to be duplicated?

File: tools/vapor/vapor.xml [hunk 2]
```diff
@@ -12,18 +12,19 @@
...
+            #set r2_ext = $fastq_input.fastq2.extension
+            ln -s '$fastq_input.fastq1' fastq1.'$r1_ext' &&
+            ln -s '$fastq_input.fastq2' fastq2.'$r2_ext' &&
+        #elif str($fastq_input.fastq_input_selector) == "paired_collection"
+            #set r1_ext = $fastq_input.fastq_pairs.forward.extension
+            #set r2_ext = $fastq_input.fastq_pairs.reverse.extension
+            ln -s '$fastq_input.fastq_pairs.forward' fastq1.'$r1_ext' &&
+            ln -s '$fastq_input.fastq_pairs.reverse' fastq2.'$r2_ext' &&
+        #else
+            ln -s '$fastq_input.fastq_single' fastq_single.'$fastq_input.fastq_single.extension' &&
```
Comment (wm75) [inline tools/vapor/vapor.xml:26]:
```suggestion
            #set r1_ext = $fastq_input.fastq_single.extension
            ln -s '$fastq_input.fastq_single' fastq1.$r1_ext &&
```

File: tools/vapor/vapor.xml [hunk 3]
```diff
@@ -35,16 +36,38 @@
...
-            #end for
+            #if str($fastq_input.fastq_input_selector) == "paired"
+                fastq1.'$r1_ext'
+                fastq2.'$r2_ext'
+            #else if str($fastq_input.fastq_input_selector) == "paired_collection"
+                fastq1.'$r1_ext'
+                fastq2.'$r2_ext'
+            #else
+                fastq_single.'$fastq_input.fastq_single.extension'
+            #end if
```
Comment (wm75) [inline tools/vapor/vapor.xml:47]:
```suggestion
            fastq1.$r1_ext
            #if str($fastq_input.fastq_input_selector) in ["paired", "paired_collection"]
                fastq2.'$r2_ext'
            #end if
```

File: tools/vapor/vapor.xml [hunk 4]
```diff
@@ -12,18 +12,19 @@
...
-                #set $ext='.fastq'
-            #end if
-            #set $out = './fastq_files/input_%s%s' % ($i, $ext)
-            ln -s '${fastq}' $out &&
-            $fastq_files.append($out)
-        #end for
+        #if str($fastq_input.fastq_input_selector) == "paired"
+            #set r1_ext = $fastq_input.fastq1.extension
+            #set r2_ext = $fastq_input.fastq2.extension
+            ln -s '$fastq_input.fastq1' fastq1.'$r1_ext' &&
```
Comment (wm75) [inline tools/vapor/vapor.xml:18]: You don't need quotes around any of the extensions cause the only possible values "fastqsanger" and "fastqsanger.gz" are unproblematic for the shell.

Comment (kciy) [reply tools/vapor/vapor.xml:18]: OK, didn't know

**Pull request merged**